UserGuide.md

```diff
@@ -260,6 +261,36 @@ gson.fromJson(json, fooType);
 ```
 The idiom used to get `fooType` actually defines an anonymous local inner class containing a method `getType()` that returns the fully parameterized type.
 
+### <a name="TOC-Parsing-a-Collection-with-Objects-of-Non-Arbitrary-Types"></a>Parsing a Collection with Objects of Non Arbitrary Types
+
+At times you will try to parse a collection of a specific object. The `getJson` and `fromJson`
+methods both provide a variety of parameter configurations to suit your needs. To parse a collection of a class :
+
+```json
+[
+  {
+    "name": "John",
+    "age": 25
+  },
+  {
+    "name": "Steven",
+    "age": 30
+  }
+]  //Example of json response to parse
+```
+
+`fromJson` provides the ability to add a third party class as an argument for serialization as so:
+```java
+public class Person() {  //example class
+    String name;
+    int age;
+}
+
+Gson gson = new Gson();
+Person[] people = gson.fromJson(json, Person[].class);
```

Comment:
I think it would be better if this deserialized into a `List<Person>` rather than a `Person[]`. For example as explained [here](https://www.baeldung.com/gson-list). Object arrays are generally better avoided and I don't think there's a strong reason to use one here.